### ISSUE TITLE
important functions for 3D view

### ISSUE THREAD

**Issue by andrea-bistacchi** [andrea-bistacchi/PZero#18]
We need to implement these functions of the [pyvista.Plotter](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.html) in order to:

1) properly navigate in the 3D window:

- [Plotter.enable_trackball_style](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_trackball_style.html#pyvista.Plotter.enable_trackball_style)()
- [Plotter.enable_zoom_style](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_zoom_style.html#pyvista.Plotter.enable_zoom_style)()
- [Plotter.fly_to](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.fly_to.html#pyvista.Plotter.fly_to)(point)
- [Plotter.fly_to_mouse_position](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.fly_to_mouse_position.html#pyvista.Plotter.fly_to_mouse_position)([focus])
- [Plotter.increment_point_size_and_line_width](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.increment_point_size_and_line_width.html#pyvista.Plotter.increment_point_size_and_line_width)(...)
- [Plotter.isometric_view](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.isometric_view.html#pyvista.Plotter.isometric_view)()
- [Plotter.remove_legend](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.remove_legend.html#pyvista.Plotter.remove_legend)([render])
- [Plotter.remove_scalar_bar](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.remove_scalar_bar.html#pyvista.Plotter.remove_scalar_bar)([title, render])
- [Plotter.reset_camera](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.reset_camera.html#pyvista.Plotter.reset_camera)([render, bounds])
- [Plotter.screenshot](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.screenshot.html#pyvista.Plotter.screenshot)([filename, ...])
- [Plotter.set_background](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.set_background.html#pyvista.Plotter.set_background)(color[, top, ...])
- [Plotter.set_focus](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.set_focus.html#pyvista.Plotter.set_focus)(point)
- [Plotter.show_grid](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.show_grid.html#pyvista.Plotter.show_grid)(**kwargs)
- [Plotter.update_scalar_bar_range](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.update_scalar_bar_range.html#pyvista.Plotter.update_scalar_bar_range)(clim[, name])
- [Plotter.update_scalars](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.update_scalars.html#pyvista.Plotter.update_scalars)(scalars[, mesh, render])
- [Plotter.view_isometric](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.view_isometric.html#pyvista.Plotter.view_isometric)([negative])

2) export 3D window:

- [Plotter.export_gltf](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_gltf.html#pyvista.Plotter.export_gltf)(filename[, inline_data, ...])
- [Plotter.export_html](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_html.html#pyvista.Plotter.export_html)(filename[, backend])
- [Plotter.export_obj](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_obj.html#pyvista.Plotter.export_obj)(filename)
- [Plotter.to_pythreejs](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.to_pythreejs.html#pyvista.Plotter.to_pythreejs)()

3) Picking in 3D window:

- [Plotter.get_pick_position](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.get_pick_position.html#pyvista.Plotter.get_pick_position)()
- [Plotter.key_press_event](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.key_press_event.html#pyvista.Plotter.key_press_event)(obj, event)
- [Plotter.left_button_down](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.left_button_down.html#pyvista.Plotter.left_button_down)(obj, event_type)
- [Plotter.pick_click_position](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.pick_click_position.html#pyvista.Plotter.pick_click_position)()
- [Plotter.pick_mouse_position](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.pick_mouse_position.html#pyvista.Plotter.pick_mouse_position)()
- [Plotter.store_click_position](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.store_click_position.html#pyvista.Plotter.store_click_position)(*args)
- [Plotter.store_mouse_position](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.store_mouse_position.html#pyvista.Plotter.store_mouse_position)(*args)

**Comment by gbene** [comment]
Yes! I would also add a .gif exporting option (https://docs.pyvista.org/examples/02-plot/gif.html) in combination with the camera orbiting function (https://docs.pyvista.org/examples/02-plot/orbit.html) to quickly create 360 degree views of the models.

**Comment by andrea-bistacchi** [comment]
Yes!

**Comment by andrea-bistacchi** [comment]
Implemented on 2/11/2022:

- [Plotter.export_gltf](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_gltf.html#pyvista.Plotter.export_gltf)()

- [Plotter.export_html](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_html.html#pyvista.Plotter.export_html)()

- [Plotter.export_obj](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_obj.html#pyvista.Plotter.export_obj)()

- [Plotter.export_vtkjs](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_vtkjs.html)()

GLTF seems to work but must be tested, HTML works but results in a very simplified 3D viewer and there are issues with textured images, OBJ works with issues for color and textured images (but this could be a limit of the viewer), and VTKjs is difficult to test.

**Comment by gbene** [comment]
GLTF files can be visualized with https://gltf-viewer.donmccurdy.com/. There are some problems related to this method. 
1. It does not work if the 3D orientation widget is active (can be easily fixed) 
2. For some reason the backsides of planes are not visualized (see video). I checked and it seems to be a VTK feature/problem.




https://user-images.githubusercontent.com/40857804/217778862-b6db8b52-92c2-407d-af1e-50c446aa6d44.mp4




VTKjs can be checked with http://viewer.pyvista.org/. The export seems to work but color bars are not exported
![image](https://user-images.githubusercontent.com/40857804/217764560-34444755-ae95-4ab9-bf88-8ef5bb894ffb.png)
![image](https://user-images.githubusercontent.com/40857804/217765082-0546bda1-9eba-4b9c-8f5b-9996f17b7625.png)

**Comment by andrea-bistacchi** [comment]
We can leave this in a lower priority.
